src/getTreeColumnKey.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Cube,
+  DataVisualizationWidgetMapping,
+  getHierarchy,
+  getLevel,
+} from "@activeviam/activeui-sdk";
+import { quote } from "@activeviam/mdx";
+
+/**
+ * Returns the column key to be used for mapping column widths to their respective columns.
+ */
+export function getTreeColumnKey({
+  mapping,
+  cube,
+}: {
+  mapping: DataVisualizationWidgetMapping;
+  cube: Cube;
+}): string {
+  switch (mapping.rows[0].type) {
+    case "hierarchy": {
+      const hierarchy = mapping.rows[0]
+        ? getHierarchy(
+            {
+              dimensionName: mapping.rows[0].dimensionName,
+              hierarchyName: mapping.rows[0].hierarchyName,
+            },
+            cube,
+          )
+        : undefined;
+
+      const firstLevelName =
+        hierarchy && hierarchy.levels[hierarchy.slicing ? 0 : 1];
+      const firstLevel =
+        firstLevelName && mapping.rows[0]
+          ? getLevel(
+              {
+                dimensionName: mapping.rows[0].dimensionName,
+                hierarchyName: mapping.rows[0].hierarchyName,
+                levelName: firstLevelName.name,
+              },
+              cube,
+            )
+          : undefined;
+
+      const { dimensionName, hierarchyName } = mapping.rows[0];
+
+      const levelName = firstLevel
+        ? firstLevel.name
+        : mapping.rows[0].levelName;
+
+      return quote(dimensionName, hierarchyName, levelName);
```

Comment:
@Nouzbe Indeed it can and this is actually what I was referring to [here](https://github.com/activeviam/atoti-ui-migration/pull/104#discussion_r1294170228). If you have the first column representing a level that is **not** the first level of the hierarchy (see image below), `bookId` for example, instead of the column key being `[Booking].[Desk].[BookId]`, AUI 5 expects it to be `[Booking].[Desk].[LegalEntity] ([Dimension].[Hierarchy].[firstLevelOfHierarchy])`, which is why I've added a bit more logic to make sure the column key ends up being what AUI 5 expects. 

![Screenshot 2023-08-22 at 4 09 59 PM](https://github.com/activeviam/atoti-ui-migration/assets/6680596/f18cafc7-8b5c-43ed-b41d-d742181f4cf1)